core/src/main/kotlin/org/evomaster/core/output/PartialOracles.kt

```diff
@@ -71,4 +76,45 @@ class PartialOracles {
         else return false;
     }
 
+    fun generatesExpectation(individual: EvaluatedIndividual<RestIndividual>): Boolean{
+        return oracles.any { oracle ->
+            individual.evaluatedActions().any {
+                oracle.generatesExpectation(
+                        (it.action as RestCallAction),
+                        (it.result as RestCallResult)
+                )
+            }
+        }
+    }
+
+    fun failByOracle(individuals: MutableList<EvaluatedIndividual<RestIndividual>>): MutableMap<String, MutableList<EvaluatedIndividual<RestIndividual>>>{
```

Comment:
unclear what this method is for. add some JavaDoc.
furthermore, do we need `Mutable` data structures here for `individuals`? are you modiyfing it by side-effects?

Reply:
also explain what is the Key/Value in the returned map